src/main/java/uk/gov/pay/api/service/ConnectorUriGenerator.java

```diff
@@ -72,7 +71,29 @@ private boolean isDirectDebitAccount(Account account) {
     }
 
     public String cancelURI(Account account, String paymentId) {
-        String path = String.format("/v1/api/accounts/%s/charges/%s/cancel", account.getAccountId(), paymentId);
-        return buildConnectorUri(account, path, Collections.emptyList()).toString();
+        String path = format("/v1/api/accounts/%s/charges/%s/cancel", account.getAccountId(), paymentId);
+        return buildConnectorUri(account, path, Maps.newHashMap());
+    }
+
+    public String eventsURI(Account account, ZonedDateTime beforeDate, ZonedDateTime afterDate, Integer page, Integer pageSize, String agreementId, String paymentId) {
+
+        Map<String, String> params = new LinkedHashMap<>();
+        
+        if (beforeDate != null)
+            params.put("before", beforeDate.format(DateTimeFormatter.ISO_INSTANT));
```

Comment:
I think we generally use `DateTimeUtils.toUTCDateString(ZonedDateTime)` for these sorts of things

Reply:
Ok, changed.